extensions/couchbase/controllerservices/CouchbaseClusterService.cpp

```diff
@@ -86,19 +87,60 @@ nonstd::expected<CouchbaseUpsertResult, CouchbaseErrorType> CouchbaseClient::ups
-    const uint16_t partition_id = (upsert_resp.mutation_token().has_value() ? upsert_resp.mutation_token()->partition_id() : 0);
     return CouchbaseUpsertResult {
       collection.bucket_name,
       upsert_resp.cas().value(),
-      partition_uuid,
-      sequence_number,
-      partition_id
+      (upsert_resp.mutation_token().has_value() ? upsert_resp.mutation_token()->sequence_number() : 0),
+      (upsert_resp.mutation_token().has_value() ? upsert_resp.mutation_token()->partition_uuid() : 0),
+      gsl::narrow<uint16_t>(upsert_resp.mutation_token().has_value() ? upsert_resp.mutation_token()->partition_id() : 0)
     };
   }
 }
 
+nonstd::expected<CouchbaseGetResult, CouchbaseErrorType> CouchbaseClient::get(const CouchbaseCollection& collection,
+      const std::string& document_id, CouchbaseValueType return_type) {
+  auto collection_result = getCollection(collection);
+  if (!collection_result.has_value()) {
+    return nonstd::make_unexpected(collection_result.error());
+  }
+
+  ::couchbase::get_options options;
+  options.with_expiry(true);
+  auto [get_err, resp] = collection_result->get(document_id, options).get();
+  if (get_err.ec()) {
+    if (getErrorType(get_err.ec()) == CouchbaseErrorType::TEMPORARY) {
+      logger_->log_error("Failed to get document '{}' from collection '{}.{}.{}' due to timeout",
+        document_id, collection.bucket_name, collection.scope_name, collection.collection_name);
+      return nonstd::make_unexpected(CouchbaseErrorType::TEMPORARY);
+    }
+    std::string cause = get_err.cause() ? get_err.cause()->message() : "";
+    logger_->log_error("Failed to get document '{}' from collection '{}.{}.{}' with error code: '{}', message: '{}'",
+      document_id, collection.bucket_name, collection.scope_name, collection.collection_name, get_err.ec(), get_err.message());
+    return nonstd::make_unexpected(CouchbaseErrorType::FATAL);
+  } else {
+    try {
+      CouchbaseGetResult result;
+      result.bucket_name = collection.bucket_name;
+      result.cas = resp.cas().value();
+      if (return_type == CouchbaseValueType::Json) {
+        result.value = resp.content_as<::couchbase::codec::binary, ::couchbase::codec::raw_json_transcoder>();
+      } else if (return_type == CouchbaseValueType::String) {
+        result.value = resp.content_as<::couchbase::codec::raw_string_transcoder>();
+      } else {
+        result.value = resp.content_as<::couchbase::codec::raw_binary_transcoder>();
+      }
+      if (resp.expiry_time().has_value()) {
+        result.expiry = utils::timeutils::getTimeStr(*resp.expiry_time());
+      }
+      return result;
+    } catch(const std::exception& ex) {
+      logger_->log_error("Failed to get content for document '{}' from collection '{}.{}.{}' with the following exception: '{}'",
+        document_id, collection.bucket_name, collection.scope_name, collection.collection_name, ex.what());
```

Comment:
```suggestion
      logger_->log_error("Failed to get content for document '{}' from collection '{}.{}.{}' with the following exception: '{}'",
          document_id, collection.bucket_name, collection.scope_name, collection.collection_name, ex.what());
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1881/commits/de69b7fd973b0333b18c97b5764396ca0fe77fe0